libbeat/beat/beat.go

```diff
@@ -97,564 +29,27 @@ type Beater interface {
 	Stop()
 }
 
-// Creator initializes and configures a new Beater instance used to execute
-// the beat its run-loop.
-type Creator func(*Beat, *common.Config) (Beater, error)
-
-// SetupMLCallback can be used by the Beat to register MachineLearning configurations
-// for the enabled modules.
-type SetupMLCallback func(*Beat) error
-
 // Beat contains the basic beat data and the publisher client used to publish
 // events.
 type Beat struct {
-	Info      common.BeatInfo // beat metadata.
-	RawConfig *common.Config  // Raw config that can be unpacked to get Beat specific config data.
-	Config    BeatConfig      // Common Beat configuration data.
-	Publisher beat.Pipeline   // Publisher pipeline
+	Info      Info     // beat metadata.
+	Publisher Pipeline // Publisher pipeline
 
 	SetupMLCallback SetupMLCallback // setup callback for ML job configs
 	InSetupCmd      bool            // this is set to true when the `setup` command is called
+
+	// XXX: remove Config from public interface
```

Comment:
It would be nice to include in the comment what's blocking this from happening or when it can be done.